.github/workflows/codeql-daily.yml

```diff
@@ -21,27 +23,151 @@ jobs:
           distribution: temurin
           java-version: 17
 
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
       - name: Initialize CodeQL
         uses: github/codeql-action/init@v3
         with:
-          languages: java, c-cpp
+          languages: java
+
+      - name: Build Java code
+        run: ./gradlew assemble --no-build-cache
+        # Skip build cache for full code analysis
+
+      - name: Perform CodeQL analysis
+        uses: github/codeql-action/analyze@v3
+        with:
+          category: java
+
+  # ===== C++ Analysis Job =====
+  analyze-cpp:
+    name: "Analyze C++ Code"
+    permissions:
+      actions: read
+      security-events: write
+    runs-on: windows-latest
+    
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Set up Java 17 (required for JNI compilation)
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 17
+
+      - name: Setup Visual Studio Build Tools
+        uses: microsoft/setup-msbuild@v1
+      
+      - name: Set up Windows SDK
+        uses: ilammy/msvc-dev-cmd@v1
```

Comment:
is there anything more official for this?

Reply:
Done